### PR TITLE
Update Envoy compatibility matrices to include consul 1.20.x and dataplane 1.6.x

### PR DESCRIPTION
**Pull request by nathancoleman**
### Description
Update Envoy matrices to include latest set of releases for consul + consul-dataplane

### PR Checklist

* [ ] updated test coverage
* [x] external facing docs updated
* [x] appropriate backport labels added
* [x] not a security concern

### CODE REVIEW

File: website/content/docs/connect/proxies/envoy.mdx [hunk 1]
```diff
@@ -122,7 +123,6 @@ until the LTS release reaches its end of maintenance.
 
 | Consul Version | Default `consul-dataplane` Version   | Other compatible `consul-dataplane` Versions |
 | -------------- | -------------------------------------|----------------------------------------------|
-| 1.19.x Ent     | 1.5.x (Envoy 1.29.x)                 | 1.4.x (Envoy 1.28.x)                         |
```
Comment (nathancoleman) [inline website/content/docs/connect/proxies/envoy.mdx:125]: I'm not sure why 1.19 was included in this table of LTS releases, but I removed it

File: website/content/docs/connect/proxies/envoy.mdx [hunk 2]
```diff
@@ -108,6 +108,7 @@ apply to both Consul Enterprise and Consul community edition (CE).
 
 | Consul Version | Default `consul-dataplane` Version   | Other compatible `consul-dataplane` Versions |
 | -------------- | -------------------------------------|----------------------------------------------|
+| 1.20.x CE      | 1.6.x (Envoy 1.31.x)                 | 1.5.x (Envoy 1.29.x)                         |
 | 1.19.x CE      | 1.5.x (Envoy 1.29.x)                 | 1.4.x (Envoy 1.28.x)                         |
```
Comment (nathancoleman) [inline website/content/docs/connect/proxies/envoy.mdx:112]: Interestingly, 1.20.x includes Envoy 1.31.x ([link](https://github.com/hashicorp/consul/blob/release/1.20.x/envoyextensions/xdscommon/ENVOY_VERSIONS)) while 1.19.x never made it to 1.30.x and stayed at 1.29.x ([link](https://github.com/hashicorp/consul/blob/release/1.19.x/envoyextensions/xdscommon/ENVOY_VERSIONS)), but this table reflects current reality

**Review by nathancoleman — COMMENTED** [review]
Personal review

File: website/content/docs/connect/proxies/envoy.mdx [hunk 3]
```diff
@@ -136,6 +136,7 @@ security fixes.
 
 | `consul-dataplane` Version Range | Associated Consul Enterprise version     | Contained Envoy Binary Version |
 | -------------------------------- | ---------------------------------------- | ------------------------------ |
+| 1.6.0 - 1.6.latest               | 1.20.x Ent                               | Envoy 1.31.x                   |
```
Comment (zalimeni) [inline website/content/docs/connect/proxies/envoy.mdx:139]: We should probably correct the second column below as well; it looks like starting w/ 1.2.x we conflated "Ent" w/ "Ent LTS"

File: website/content/docs/connect/proxies/envoy.mdx (hunk 1 above)
Comment (zalimeni) [reply website/content/docs/connect/proxies/envoy.mdx:125]: 🍝

File: website/content/docs/connect/proxies/envoy.mdx (hunk 2 above)
Comment (zalimeni) [reply website/content/docs/connect/proxies/envoy.mdx:112]: We had to jump two versions at some point bc we'd started to lag >1 release on latest Envoy vs. latest Consul due to release cadence differences. 1.20 happened to be selected as the version for the jump.

File: website/content/docs/connect/proxies/envoy.mdx (hunk 3 above)
Comment (nathancoleman) [reply website/content/docs/connect/proxies/envoy.mdx:139]: True, it's easy to get "active versions of Consul" confused with "LTS versions of Consul"

Comment (nathancoleman) [reply website/content/docs/connect/proxies/envoy.mdx:139]: Done

File: website/content/docs/connect/proxies/envoy.mdx [hunk 4]
```diff
@@ -136,10 +136,6 @@ security fixes.
 
 | `consul-dataplane` Version Range | Associated Consul Enterprise version     | Contained Envoy Binary Version |
 | -------------------------------- | ---------------------------------------- | ------------------------------ |
-| 1.5.0 - 1.5.latest               | 1.18.x Ent                               | Envoy 1.29.x                   |
-| 1.4.0 - 1.4.latest               | 1.18.x Ent                               | Envoy 1.28.x                   |
```
Comment (nathancoleman) [inline website/content/docs/connect/proxies/envoy.mdx:140]: I've removed these as neither dataplane release spans multiple major Envoy versions, which is what this section of the doc is meant to be

File: website/content/docs/connect/proxies/envoy.mdx [hunk 5]
```diff
@@ -136,10 +136,6 @@ security fixes.
...
-| 1.2.0 - 1.2.latest               | 1.18.x Ent                               | Envoy 1.27.x                   |
-| 1.2.0 - 1.2.7                    | 1.18.x Ent                               | Envoy 1.27.x                   |
```
Comment (nathancoleman) [inline website/content/docs/connect/proxies/envoy.mdx:142]: I've removed these as they align with a now-unsupported version of Consul

**Review by zalimeni — APPROVED** [review]

**Pull request merged**